gapis/api/vulkan/image_primer.go

```diff
@@ -48,286 +48,30 @@ const (
-	if queue.IsNil() {
-		return log.Err(p.sb.ctx, nil, "[Priming image data by pre-initialization] Nil queue")
-	}
-	newImg := GetState(p.sb.newState).Images().Get(img.VulkanHandle())
-	// TODO: Handle multi-planar images
-	newImgPlaneMemInfo, _ := subGetImagePlaneMemoryInfo(p.sb.ctx, nil, api.CmdNoID, nil, p.sb.newState, GetState(p.sb.newState), 0, nil, nil, newImg, VkImageAspectFlagBits(0))
-	newMem := newImgPlaneMemInfo.BoundMemory()
-	oldImgPlaneMemInfo, _ := subGetImagePlaneMemoryInfo(p.sb.ctx, nil, api.CmdNoID, nil, p.sb.oldState, GetState(p.sb.oldState), 0, nil, nil, img, VkImageAspectFlagBits(0))
-	boundOffset := oldImgPlaneMemInfo.BoundMemoryOffset()
-	planeMemRequirements := oldImgPlaneMemInfo.MemoryRequirements()
-	boundSize := planeMemRequirements.Size()
-	dat := p.sb.MustReserve(uint64(boundSize))
-
-	at := NewVoidᵖ(dat.Ptr())
-	atdata := p.sb.newState.AllocDataOrPanic(p.sb.ctx, at)
-	p.sb.write(p.sb.cb.VkMapMemory(
-		newMem.Device(),
-		newMem.VulkanHandle(),
-		boundOffset,
-		boundSize,
-		VkMemoryMapFlags(0),
-		atdata.Ptr(),
-		VkResult_VK_SUCCESS,
-	).AddRead(atdata.Data()).AddWrite(atdata.Data()))
-	atdata.Free()
-
-	transitionInfo := []imageSubRangeInfo{}
-	for _, rng := range opaqueBoundRanges {
-		walkImageSubresourceRange(p.sb, img, rng,
-			func(aspect VkImageAspectFlagBits, layer, level uint32, unused byteSizeAndExtent) {
-				origLevel := img.Aspects().Get(aspect).Layers().Get(layer).Levels().Get(level)
-				origDataSlice := origLevel.Data()
-				linearLayout := origLevel.LinearLayout()
-
-				p.sb.ReadDataAt(origDataSlice.ResourceID(p.sb.ctx, p.sb.oldState), uint64(linearLayout.Offset())+dat.Address(), origDataSlice.Size())
-
-				transitionInfo = append(transitionInfo, imageSubRangeInfo{
-					aspectMask:     VkImageAspectFlags(aspect),
-					baseMipLevel:   level,
-					levelCount:     1,
-					baseArrayLayer: layer,
-					layerCount:     1,
-					oldLayout:      VkImageLayout_VK_IMAGE_LAYOUT_PREINITIALIZED,
-					newLayout:      origLevel.Layout(),
-					oldQueue:       queue.VulkanHandle(),
-					newQueue:       queue.VulkanHandle(),
-				})
-			})
-	}
-
-	p.sb.write(p.sb.cb.VkFlushMappedMemoryRanges(
-		newMem.Device(),
-		1,
-		p.sb.MustAllocReadData(NewVkMappedMemoryRange(p.sb.ta,
-			VkStructureType_VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE, // sType
-			0,                     // pNext
-			newMem.VulkanHandle(), // memory
-			0,                     // offset
-			boundSize,             // size
-		)).Ptr(),
-		VkResult_VK_SUCCESS,
-	))
-	dat.Free()
-
-	p.sb.write(p.sb.cb.VkUnmapMemory(
-		newMem.Device(),
-		newMem.VulkanHandle(),
-	))
-
-	p.sb.changeImageSubRangeLayoutAndOwnership(img.VulkanHandle(), transitionInfo)
-
-	return nil
-}
-
 func (p *imagePrimer) free() {
 	p.rh.free()
 	p.sh.free()
 }
 
 // internal functions of image primer
 
-func (p *imagePrimer) allocStagingImages(img ImageObjectʳ, aspect VkImageAspectFlagBits) ([]ImageObjectʳ, []DeviceMemoryObjectʳ, error) {
```

Comment:
This function has gotten complicated enough, that it really deserves a more verbose comment.
At the least, what the returns are. I do not know what that func() is supposed to be.

Reply:
Comments added in #2558, also this function got renamed to `create32BitUintColorStagingImagesForAspect()`